### ISSUE TITLE
Python3 error when plotting graph inline in jupyter notebook

### ISSUE THREAD

**Issue by vtraag** [igraph/python-igraph#88]
Executing 
```python
G = ig.Graph.Erdos_Renyi(100, 0.1);
ig.plot(G)
```
yields the error
```python
AttributeError: 'bytes' object has no attribute 'encode'
```
Using igraph version 0.7.1.post6, python 3.5.2, ipython 5.0.0, jupyter server 4.2.2.

**Comment by ntamas** [comment]
Probably resolved in https://github.com/igraph/python-igraph/commit/abb4a67b24b9ee8efcfb1629cc6c58358cc306c2 and/or in https://github.com/igraph/python-igraph/commit/3eb9f27139eb8d7dc2d7416181b03f6f6cd83862 .

**Comment by ntamas** [comment]
Also updated in the 0.7 release branch in 8864b468.

**Comment by foice** [comment]
Having the same trouble with igraph version 0.7.1.post6, python 3.6, ipython 5.1.0, jupyter server 4.3.1

        ---------------------------------------------------------------------------
        AttributeError                            Traceback (most recent call last)
        /usr/local/lib/python3.6/site-packages/IPython/core/formatters.py in __call__(self, obj)
            309             method = get_real_method(obj, self.print_method)
            310             if method is not None:
        --> 311                 return method()
            312             return None
            313         else:

        /usr/local/lib/python3.6/site-packages/igraph/drawing/__init__.py in _repr_svg_(self)
            352         surface.finish()
            353         # Return the raw SVG representation
        --> 354         return io.getvalue().encode("utf-8")
            355 
            356     @property

        AttributeError: 'bytes' object has no attribute 'encode'




Any idea what to do to resolve?

**Comment by ntamas** [comment]
Please apply the fix in https://github.com/igraph/python-igraph/commit/8864b46849b031a3013764d03e167222963c0f5d manually (there was no new release since the fix was committed).

**Comment by foice** [comment]
Downloading from 
`https://github.com/igraph/python-igraph/archive/master.zip`
and installing using
`sudo python setup.py develop --c-core-url https://github.com/igraph/igraph/archive/master.tar.gz`

should give me the patched version? or am I misreading the code in the master?

**Comment by foice** [comment]
added `python-igraph-master/build/lib.macosx-10.12-x86_64-3.6/igraph/` to `$PYTHONPATH` and seems to work. Thanks!

**Comment by ntamas** [comment]
How about running `python setup.py install` from `python-igraph-master` instead? That should also do the trick.

**Comment by foice** [comment]
I will try at the next occasion, the command quoted above make the package, I had just to add it to the python packages path. Now dealing with plotting my 20M links graph ... but that's another story ...

**Comment by Hades1996** [comment]
it works, pip should have a newer version of python-igraph (it's still using 0.7.1.post6)

**Comment by claresloggett** [comment]
Thanks for this fix! It's definitely needed for igraph in Jupyter. Any chance of a bugfix release with this change? I think it's the easiest way for us to be able to install a working version of python-igraph via conda, which is nice as it handles all dependences?

**Comment by aopisco** [comment]
> Please apply the fix in [8864b46](https://github.com/igraph/python-igraph/commit/8864b46849b031a3013764d03e167222963c0f5d) manually (there was no new release since the fix was committed).

make a note here for newbies:
the file you need to edit is the `__init__.py` in `~/anaconda3/lib/python3.6/site-packages/igraph/drawing` and quick you to jump to the correct location is to search for *"SVG representation"*

**Comment by healiseu** [comment]
Hi @ntamas the fix has not been applied to the latest release when you install it with `pip install python-igraph`. I edited the `igraph/drawing/__init__.py` but I got the same error during the execution of `plot` command in Jupyter Notebook, I guess your module has to be recompiled. Therefore at the present the only solution seems to be installation from `python-igraph-master` as you suggest.

```text
OS Ubuntu 16.04

pip show python-igraph
Name: python-igraph
Version: 0.7.1.post6
Summary: High performance graph data structures and algorithms
Home-page: http://pypi.python.org/pypi/python-igraph

Python 3.7.3 | packaged by conda-forge | (default, Mar 27 2019, 23:01:00) 
IPython 7.5.0 -- An enhanced Interactive Python. Type '?' for help.
The version of the notebook server is: 5.7.8
```

**Comment by healiseu** [comment]
The trick to apply the following fix for  Python 3.7.3
```python
result = io.getvalue()
return result.decode("utf-8")
```
is to download the release first with

```python
pip download python-igraph
```
Unpack edit the file `igraph/drawing/__init__.py` and apply the fix, then install 
```python
pip install  python-igraph-0.7.1.post6.tar.gz
```

**Comment by vtraag** [comment]
FYI, the version on the conda-forge channel of Anaconda https://anaconda.org/conda-forge/python-igraph does include this fix. It simply compiled the release branch that contains this fix, so it isn't a proper new release.

**Comment by aparcar** [comment]
Is this fix planed for a soon to come release?

**Comment by vtraag** [comment]
We are currently working on releasing a new version of the C core `igraph`. After that, we will probably try to release a new version of `python-igraph` as soon as possible. Unfortunately, we cannot promise to release on any particular date.

**Comment by ntamas** [comment]
However, note that simply running `pip install git+https://github.com/igraph/python-igraph#egg=python-igraph` should work out of the box now and install the head revision of `python-igraph` that already includes this fix.

**Comment by FanchenBao** [comment]
I had a fresh install from `pip3 install python-igraph`, had the same problem as described here, and fixed it using the manual method described here https://github.com/igraph/python-igraph/issues/88#issuecomment-275946658

**Comment by ntamas** [comment]
For what it's worth, igraph 0.8 was released yesterday and python-igraph 0.8 will follow it in a few days - this new version will include the fix.
